src/rpm.rs

```diff
@@ -45,7 +46,43 @@ impl RPMPackage {
         out.write_all(&self.content)?;
         Ok(())
     }
+
+    /// sign all headers (except for the lead) using an external key and store it as the initial header
+    pub fn sign<S: pgp::types::SecretKeyTrait>(&mut self, secret_key: &S) -> Result<(), RPMError> {
```

Comment:
I think we should either accept something generlic like &[u8] and parse the key later, or reexport the necessary pgp types in order to make this easy to use.

Reply:
I am not a big fan of typeless parameters, I'd rather have an intermediate type which has well documented `try_from` / `parse` / `serde` impls.

Reply:
This is obsolete, with the trait based approach the signer trait will introduce a trait bound for the key, which implies convertable to DER (which seems to be a sane common ground, or pkcs15).